File: algorithms/1_easy/M/03487_maximum_unique_subarray_sum_after_deletion/solution.py
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        nums_set = set()
        max_val = -100000000000000

        for num in nums:
            if num > 0:
                nums_set.add(num)
            max_val = max(num, max_val)

        if not nums_set:
            return max_val

        return sum(nums_set)

File: algorithms/1_easy/M/03487_maximum_unique_subarray_sum_after_deletion/solution.rs
use std::collections::HashSet;

impl Solution {
    pub fn max_sum(nums: Vec<i32>) -> i32 {

        let mut nums_set: HashSet<i32> = HashSet::new();
        let mut max = i32::MIN;

        for num in nums.into_iter() {
            if num > 0 {
                nums_set.insert(num);
            }
            max = max.max(num);
        }

        if nums_set.is_empty() {
            return max;
        }

        nums_set.into_iter().sum()
    }
}
